presto-hive/src/main/java/com/facebook/presto/hive/PartitionUpdate.java

```diff
@@ -205,4 +206,60 @@ public HiveBasicStatistics getStatistics()
         APPEND,
         OVERWRITE,
     }
+
+    public static class StagingFileInfo
```

Comment:
Not sure if this we want to have other class name (e.g. `FileCommitInfo`). This class name makes me think staging file is always happening.

Similar to the field `stagingFileName` -- this field presents even we don't write to stage files. Maybe make it Optional?

Let's discuss in person.

Reply:
How about let's rename the class to `FileWriteInfo` and field/variable to `writeFileName` so that it does not indicate staging file is always happening? We used similar method in `HiveLocationService` depending on different `WriteMode`.